utils/gpi_update_wcs_basic.pro

```diff
@@ -154,7 +159,9 @@ pro gpi_update_wcs_basic,backbone,parang=parang,imsize=imsize
      if utendd lt utstartd then dateline = 1 else dateline = 0 ;account for crossing UTC dateline
      expend = utendd - 0.5d0*readtimed
                                 ;expstart = utendd - (ngroup-0.5)*readtimed
-     expstart = utendd - itime/3600d
+                                ;expstart = utendd - itime/3600d
+     expstart = utendd - readtimed*(double(ncoadd)*(double(nread)+1.0d)-3.0d/2.0d)
```

Comment:
Can you explain where the -3/2 part comes from? I don’t quite follow that...though to be fair I’m reading this first thing in the morning pre coffee.

Reply:
utendd is when the last pixel is read, so 1/2 gets you to to average pixel. Then another 2/2 is needed because we don't care about the first reset. The two ways we could implement this are

`expstart = utendd - readtimed*(double(ncoadd)*(double(nread)+1.0d)-3.0d/2.0d)`

or


`expstart = expend - readtimed*(double(ncoadd)*(double(nread)+1.0d)-2.0d)`

The original version also had an error for single coadd frames that expstart was computed relative to utendd not expend. I think this means all times are out by half a read for all single coadd frames.